src/layer-composer/generators/track/track.ts

```diff
@@ -135,6 +174,11 @@ class TrackGenerator {
     memoizeByLayerId(config.id, {
       simplifyTrackWithZoomLevel: memoizeOne(simplifyTrackWithZoomLevel),
       filterByTimerange: memoizeOne(filterByTimerange),
+    })
+    memoizeByLayerId(`${config.id}-${this.highlightSufix}`, {
```

Comment:
Not sure this is right. Wouldn't it be better to memoize by `config.id`, and have two proxy functions that redirect to to `getHighlightedData` ?
```
const getHighlightedData = (...) => getTrackPortion(...)
const getHighlightedEventData = (...) => getTrackPortion(...)

memoizeByLayerId(config.id, {
  ...
  getHighlightedData: memoizeOne(getHighlightedData),
  getHighlightedEventData: memoizeOne(getHighlightedEventData),
})

Reply:
Your way is way simpler... I don't know why I didn't think about it before 🤦 addressed here:https://github.com/GlobalFishingWatch/layer-composer/commit/c527c03d1ad1842e6de16554622c06a9a61d3071